src/algebraic_topology/simplicial_object.lean

```diff
@@ -435,6 +458,28 @@ variable {C}
 
 end augmented
 
+open_locale simplicial
+
+/-- Augment a cosimplicial object with an object. -/
+@[simps]
+def augment (X : cosimplicial_object C) (X₀ : C) (f : X₀ ⟶ X.obj [0])
+  (w : ∀ (i : simplex_category) (g₁ g₂ : [0] ⟶ i),
+    f ≫ X.map g₁ = f ≫ X.map g₂) : cosimplicial_object.augmented C :=
+{ left := X₀,
+  right := X,
+  hom :=
+  { app := λ i, f ≫ X.map (simplex_category.const i 0),
+  naturality' := begin
+    intros i j g,
+    dsimp,
```

Comment:
i thought `dsimp`s couldn't be non-terminal too, but it's the second time I've seen this today (and there's quite a few in this file)

Reply:
`dsimp` is less problematic, because it preserves definitional equality. So most tactics (basically except `rw`) that follow `dsimp` will work, even if the goal slightly changes because of a new `simp`-lemma.